### PR TITLE
docs: Recommended quickstart changes + Upgrade

### PR DESCRIPTION
**Pull request by Jayclifford345**
These changes were made to fix some of the issues found in: https://github.com/grafana/loki/issues/13725

As well as this a general clean-up of Sandbox syntax so the markdown is less convoluted. I have also updated the Quickstart to Loki 3.1 and changed Alloy back to production due to a fix in 1.2. 

Sandbox in dev: https://killercoda.com/grafana-dev-testing/course/loki/loki-quickstart
(Note that due to the demo pulling from the main branch of Loki, you will not see the newly updated containers for Loki or Alloy). I have tested these locally. If you would like to test within the sandbox you can run this command when pulling the docker-compose"
```bash
wget https://raw.githubusercontent.com/grafana/loki/recommended-quickstart-changes/examples/getting-started/loki-config.yaml -O loki-config.yaml
wget https://raw.githubusercontent.com/grafana/loki/recommended-quickstart-changes/examples/getting-started/alloy-local-config.yaml -O alloy-local-config.yaml
wget https://raw.githubusercontent.com/grafana/loki/recommended-quickstart-changes/examples/getting-started/docker-compose.yaml -O docker-compose.yaml
```

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: docs/sources/get-started/quick-start.md [hunk 1]
```diff
@@ -48,34 +45,22 @@ The Docker Compose configuration runs the following components, each in its own
...
 {{< /admonition >}}
 
-Try out this demo within our interactive learning environment: [Loki Quickstart Sandbox](https://killercoda.com/grafana-labs/course/loki/loki-quickstart)
-
-- You must have a free Killercoda account to verify you aren't a bot.
-- Tutorial instructions are located on the left-side of the screen.
-  Click to move on to the next section.
-- All commands run inside the interactive terminal.
-- You can access Grafana with the URL links provided within the sandbox.
+- Install [Docker](https://docs.docker.com/install)
```
Comment (JStickler) [inline docs/sources/get-started/quick-start.md:61]: I would move these two bullet points above the Tip admonition.  Right now the admonition reads "Alternatively, you can try out...." but I think it makes more sense for that to come **_after_** we tell users they have to install Docker and Docker Compose.

File: docs/sources/get-started/quick-start.md [hunk 2]
```diff
@@ -316,12 +290,66 @@ To see every log line that doesn't contain the text `401`:
...
+        EOF
+        /run.sh
+```
+Within the entrypoint section, the Loki data source is configured with the following details:
+- Name: Loki (name of the data source)
+- Type: loki (type of data source)
+- Access: proxy (access type)
+- URL: http://gateway:3100 (URL of the Loki data source. Loki uses a nginx gateway to direct traffic to the appropriate component)
+- jsonData: httpHeaderName1: "X-Scope-OrgID" (header name for the organization ID)
+- secureJsonData: httpHeaderValue1: "tenant1" (header value for the organization ID)
```
Comment (JStickler) [inline docs/sources/get-started/quick-start.md:331]:
```suggestion
- `Name: Loki` (name of the data source)
- `Type: loki` (type of data source)
- `Access: proxy` (access type)
- `URL: http://gateway:3100` (URL of the Loki data source. Loki uses an nginx gateway to direct traffic to the appropriate component)
- `jsonData.httpHeaderName1: "X-Scope-OrgID"` (header name for the organization ID)
- `secureJsonData.httpHeaderValue1: "tenant1"` (header value for the organization ID)
```

File: docs/sources/get-started/quick-start.md [hunk 3]
```diff
@@ -316,12 +290,66 @@ To see every log line that doesn't contain the text `401`:
...
+  
+It is important to note when Loki is configured in any other mode other than monolithic deployment, a tenant ID is required to be passed in the header. Without this, queries will return an authorization error.
```
Comment (JStickler) [inline docs/sources/get-started/quick-start.md:333]:
```suggestion
It is important to note when Loki is configured in any other mode other than monolithic deployment, you are required to pass a tenant ID in the header. Without this, queries will return an authorization error.
```

File: docs/sources/get-started/quick-start.md [hunk 4]
```diff
@@ -316,12 +290,66 @@ To see every log line that doesn't contain the text `401`:
...
 
+<!-- INTERACTIVE page finish.md START -->
+
 ## Complete metrics, logs, traces, and profiling example
 
+You have completed the Loki Quickstart demo. So where to go next?
+
+{{< docs/ignore >}}
+## Back to docs
+Head back to wear you started from to continue with the Loki documentation: [Loki documentation](https://grafana.com/docs/loki/latest/get-started/quick-start/).
```
Comment (JStickler) [inline docs/sources/get-started/quick-start.md:345]:
```suggestion
Head back to where you started from to continue with the Loki documentation: [Loki documentation](https://grafana.com/docs/loki/latest/get-started/quick-start/).
```

File: docs/sources/get-started/quick-start.md (hunk 1 above)
Comment (Jayclifford345) [reply docs/sources/get-started/quick-start.md]: Great catch! Should be updated now

**Review by slim-bean — APPROVED** [review]

**Pull request merged**